modules/utxo-lib/src/bitgo/zcash/ZcashPsbt.ts

```diff
@@ -22,6 +26,72 @@ export class ZcashPsbt extends UtxoPsbt<ZcashTransaction<bigint>> {
     );
   }
 
+  /**
+   * In version < 5 of Zcash transactions, the consensus branch ID is not serialized in the transaction
+   * whereas in version 5 it is. If the transaction is less than a version 5, set the consensus branch id
+   * in the global map in the psbt. If it is a version 5 transaction, throw an error if the consensus
+   * branch id is set in the psbt (because it should be on the transaction already).
+   * @param buffer Psbt buffer
+   * @param opts options
+   */
+  static fromBuffer(buffer: Buffer, opts: PsbtOpts): UtxoPsbt<UtxoTransaction<bigint>> {
+    const psbt = super.fromBuffer(buffer, opts) as ZcashPsbt;
+
+    // Read `consensusBranchId` from the global-map
+    const consensusIdKey = Buffer.concat([
+      Buffer.of(0xfc),
+      Buffer.of(0x05),
+      Buffer.from('BITGO', 'ascii'),
+      Buffer.of(Subtype.ZEC_CONSENSUS_BRANCH_ID),
+    ]);
+    let consensusBranchId: number | undefined = undefined;
+    psbt.data.globalMap.unknownKeyVals?.forEach(({ key, value }) => {
+      if (key.equals(consensusIdKey)) {
+        consensusBranchId = value.readUint32BE();
+      }
+    });
+    switch (psbt.tx.version) {
+      case 4:
+      case ZcashTransaction.VERSION4_BRANCH_CANOPY:
+      case ZcashTransaction.VERSION4_BRANCH_NU5:
+        if (!consensusBranchId) {
+          throw new Error('Could not find consensus branch id on psbt for version 4 Zcash transaction');
+        }
+        psbt.tx.consensusBranchId = consensusBranchId;
+        return psbt;
+      case 5:
+      case ZcashTransaction.VERSION5_BRANCH_NU5:
+        if (consensusBranchId) {
+          throw new Error('Found consensus branch id in psbt global-map for version 5 Zcash transaction');
+        }
+        return psbt;
+    }
+    throw new Error(`Unsupported transaction version ${psbt.tx.version}`);
+  }
+
+  /**
+   * If it is a version 4 transaction, add the consensus branch id to
+   * the global map. If it is a version 5 transaction, just return the
+   * buffer because the consensus branch id is already serialized in
+   * the transaction.
+   */
+  toBuffer(): Buffer {
+    if (this.tx.version === 5 || this.tx.version === ZcashTransaction.VERSION5_BRANCH_NU5) {
+      return super.toBuffer();
+    }
+    const key = Buffer.concat([
+      Buffer.of(0xfc),
+      Buffer.of(0x05),
+      Buffer.from('BITGO', 'ascii'),
+      Buffer.of(Subtype.ZEC_CONSENSUS_BRANCH_ID),
+    ]);
+    const value = Buffer.from(this.tx.consensusBranchId.toString(16), 'hex');
```

Comment:
I think we should use writeUint32BE (as we do for reading), now that I think about this more. This method could create a smaller than 4-byte buffer if the consensusBranchId were ever to have  leading 0 byte.